### PR TITLE
Add valueDate- based cash flow filtering to FRA via GetCFs(Date) helper method

### LINKED ISSUE #26: Refactor FRA
**Context**
The FRA product currently returns all cashflows regardless of the valuation date. According to the ProductWrapper contract, GetCFs(Date valueDate) should only return future cashflows. This update ensures consistency with other products and aligns with the intended filtering behaviour.

**Details**
The FRA doesn't have filtering logic on valueDate like some other products do, and the ProductWrapper XML documentation specifies that products GetCfs method should perform this filtering.

### PR DESCRIPTION
**Pull request by rwx-oberholzer-marcelle**
✅ **Pull Request Checklist**

### 🔍 Self-Review

* [x] Destination branch is correct.
* [x] Code has been self-reviewed and is scoped to the PR’s purpose.
* [x] No debugging code, print statements, or unrelated changes.
* [x] Code is clean, well-commented, and follows project standards.
* [x] Documentation is updated, if applicable.

### 🧪 Testing & Build

* [ ] All relevant tests (unit/integration) pass locally.
* [ ] Edge cases and error handling are considered.
* [ ] No new warnings or errors in the build or test output.

### 📄 Clarity

* [ ] PR title and description are clear and descriptive.
* [ ] Purpose and context of the change are well explained.
* [ ] Related issues are referenced (e.g., `Fixes #123`).

### 👥 Collaboration

* [x] Reviewers assigned and feedback requested.
* [ ] All review comments have been addressed or responded to.

### CODE REVIEW

File: QuantSA/QuantSA.Core/Products/Rates/FRA.cs [hunk 1]
```diff
@@ -92,5 +92,10 @@ public override List<Cashflow> GetCFs()
             if (!_payFixed) amount = -1.0 * amount;
             return new List<Cashflow> {new Cashflow(_nearDate, amount, _floatIndex.Currency) };
         }
+
+        public List<Cashflow> GetCFs(Date valueDate)
+        {
+            return _nearDate > valueDate ? GetCFs() : new List<Cashflow>();
+        }
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.Core/Products/Rates/FRA.cs:99]: Move the filter into the GetCfs() function and delete this one.

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]

File: QuantSA/QuantSA.Core/Products/Rates/FRA.cs (hunk 1 above)
Comment (rwx-oberholzer-marcelle) [reply QuantSA/QuantSA.Core/Products/Rates/FRA.cs:99]: @robertjacquier-rw, I tried to made this change but had some problems. The unit test failed after I added the date filter into the GetCFs() function.

I believe this is because the GetCFs() method is a required override from the ProductWrapper class, which expects all product cashflows to be returned (if nearDate is in the past or future). Maybe we can have a look at this on Thursday as I don't really know what to do next.

Comment (rwx-jacquier-robert) [reply QuantSA/QuantSA.Core/Products/Rates/FRA.cs:99]: @MarcelleOberholzer we'll have to chat about this so you can show me, please set up a meeting for Tues or Wed next week.

Comment (rwx-jacquier-robert) [reply QuantSA/QuantSA.Core/Products/Rates/FRA.cs:99]: @MarcelleOberholzer we didn't chat about this one, did you figure out what I meant by the first comment?

Comment (rwx-oberholzer-marcelle) [reply QuantSA/QuantSA.Core/Products/Rates/FRA.cs:99]: @robertjacquier-rw
I had another look and I now understand what you meant about moving the filter into the GetCFs() function.
So I actually tried doing that, but the test failed. I think it's because the GetCFs() method is an override from ProductWrapper, which expects all cashflows to be returned (if near date is in the past or future).
If I filter inside that method, other parts of the system that rely on seeing all cashflows might break.
So that's why I added the filtered version in this seperate GetCFs(Date valueDate) method instead.
